src/test/java/com/github/lwhite1/tablesaw/SortTest.java

```diff
@@ -1,18 +1,71 @@
 package com.github.lwhite1.tablesaw;
 
 import com.github.lwhite1.tablesaw.api.Table;
+import org.junit.ComparisonFailure;
 import org.junit.Test;
 
+import static org.junit.Assert.assertEquals;
+
 /**
- *
+ * Verify sorting functions
  */
 public class SortTest {
 
-  @Test
-  public void sort() {
-    final Table table = TestData.TORNADOES.getTable();
+    @Test
+    public void sortAscending() {
+
+        // start with unsorted data
+        Table table = TestData.SIMPLE_UNSORTED_DATA.getTable();
+
+        // sort ascending by date and then an integer
+        Table sortedTable = table.sortAscendingOn("DOB", "IQ");
+
+        // get the data that has been presorted correctly - i.e. known good results
+        Table compareWith = TestData.SIMPLE_SORTED_DATA_BY_INTEGER_AND_DATE_ASCENDING.getTable();
+
+        compareTables(sortedTable, compareWith);
+    }
+
+    /**
+     * Same as sortAscending but descending
+     */
+    @Test
+    public void sortDescending() {
+        Table table = TestData.SIMPLE_UNSORTED_DATA.getTable();
+        Table sortedTable = table.sortDescendingOn("DOB", "IQ");
+        Table compareWith = TestData.SIMPLE_SORTED_DATA_BY_INTEGER_AND_DATE_DESCENDING.getTable();
+        compareTables(sortedTable, compareWith);
+    }
+
+    /**
+     * Verify data that is not sorted descending does match data that has been
+     * (this test verifies the accuracy of our positive tests)
+     */
+    @Test(expected = ComparisonFailure.class)
+    public void sortDescendingNegative() {
+        Table table = TestData.SIMPLE_UNSORTED_DATA.getTable();
+        Table sortedTable = table.sortDescendingOn("DOB", "IQ");
+        Table compareWith = TestData.SIMPLE_SORTED_DATA_BY_INTEGER_AND_DATE_ASCENDING.getTable();
+        compareTables(sortedTable, compareWith);
+    }
 
-  }
 
+    /**
+     * Make sure each row in each table match
+     *
+     * @param sortedTable the table that was sorted with tablesaw
+     * @param compareWith the table that was sorted using some external means e.g. excel. i.e known good data
+     */
+    private void compareTables(Table sortedTable, Table compareWith) {
+        assertEquals("both tables have the same number of rows", sortedTable.rowCount(), compareWith.rowCount());
+        int maxRows = sortedTable.rowCount();
+        int numberOfColumns = sortedTable.columnCount();
```

Comment:
>> Should there be a comparison method on table? 
um. to compare as in your new compareTables() method? or for use in sorting() as in a Comparable?

>> Should there be a getRow method?
well, you get a cell with get(int rowNum, int colNumber);
There is no get(recordNumber) and there isn't one in SQL, but it's possible to add one. Generally, you would want to select a row using a query, because the table is not required to have a unique id per row (again like sql). 

When you use a query, the result is a bitmap that works like a compressed array of ints. These are the numbers of the rows you get back. It could be made user friendly by also supporting a regular int[].

If you tell me what you want, i can probably be more helpful

>>What is the purpose of the table.row(int i) method? (that confused me) 
That's a good one. Should open a bug.